catalog_reader/scripts/apps_hashes.py

```diff
@@ -67,22 +68,37 @@ def update_catalog_hashes(catalog_path: str) -> None:
             catalog_base_lib_dir_path = os.path.join(library_dir.as_posix(), lib_version)
             shutil.copytree(catalog_base_lib_dir_path, app_base_lib_dir.as_posix())
 
+            old_version = app_config['version']
+            if bump_type and app_config['lib_version_hash'] != hashes[lib_version]:
+                if not is_valid_bump_type(bump_type):
+                    print(
+                        f'[\033[91mERROR\x1B[0m]\tInvalid bump type {bump_type!r}, '
+                        f'skipping version bumping for {app_dir.name!r} in {train_dir.name}'
+                    )
+                else:
+                    app_config['version'] = bump_version(old_version, bump_type)
+                    rename_versioned_dir(old_version, app_config['version'], train_dir.name, app_dir)
+
             app_config['lib_version_hash'] = hashes[lib_version]
             with open(str(app_metadata_file), 'w') as f:
                 f.write(yaml.safe_dump(app_config))
 
-            print(f'[\033[92mOK\x1B[0m]\tUpdated library hash for {app_dir.name!r} in {train_dir.name}')
+            print(
+                f'[\033[92mOK\x1B[0m]\tUpdated library hash for {app_dir.name!r} in {train_dir.name}'
+                f' and bumped version from {old_version!r} to {app_config["version"]!r}' if bump_type else ''
+            )
 
 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('--path', help='Specify path of TrueNAS catalog')
+    parser.add_argument('--bump-type', default=None, help='Version bump type for app that the hash was updated')
```

Comment:
Okay I understand now. We can use `add_argument` here to do a bit more work for us on the front-end :smile: 

```
parser.add_argument('--bump', nargs=1, type=str, choices=('major', 'minor', 'patch'), required=False)

Reply:
Doing it this way allows us to get rid of `is_valid_bump_type`

Reply:
```
➜  middleware git:(master) ✗ python3 zz.py --bump yeah
usage: zz.py [-h] [--bump-type {major,minor,path}]
zz.py: error: argument --bump-type: invalid choice: 'yeah' (choose from 'major', 'minor', 'path')